security/rbac/scripts/enable-rbac-connect.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+
+
+################################################################################
+# Overview
+################################################################################
+#
+################################################################################
+
+# Source library
+. ../../../utils/helper.sh
+. ./rbac_lib.sh
+
+check_env || exit 1
+check_cli_v2 || exit 1
+check_jq || exit 1
+
+##################################################
+# Initialize
+##################################################
+
+. ../config/local-demo.cfg
+ORIGINAL_CONFIGS_DIR=/tmp/original_configs
+DELTA_CONFIGS_DIR=../delta_configs
+FILENAME=connect-avro-distributed.properties
+create_temp_configs $CONFLUENT_HOME/etc/schema-registry/$FILENAME $ORIGINAL_CONFIGS_DIR/$FILENAME $DELTA_CONFIGS_DIR/${FILENAME}.delta
+
+# Log in to Metadata Server (MDS)
+login_mds $MDS
+
+##################################################
+# Administrative Functions
+# - Grant the principal User:$ADMIN_CONNECT to the ResourceOwner role for Topic:connect-configs
+# - Grant the principal User:$ADMIN_CONNECT to the ResourceOwner role for Topic:connect-offsets
+# - Grant the principal User:$ADMIN_CONNECT to the ResourceOwner role for Topic:connect-statuses
+# - Grant the principal User:$ADMIN_CONNECT to the ResourceOwner role for Group:connect-cluster
+# - Grant the principal User:$ADMIN_CONNECT to the ResourceOwner role for Topic:_secrets (for Secret Registry)
+# - Grant the principal User:$ADMIN_CONNECT to the ResourceOwner role for Group:secret-registry (for Secret Registry)
+# - Grant SecurityAdmin access to the Connect cluster administrator $ADMIN_CONNECT
+##################################################
+
+# Get the Kafka cluster id
+get_cluster_id_kafka
+
+for CONNECT_TOPIC in connect-configs connect-offsets connect-statuses; do
+  echo -e "\n# Grant the principal User:$ADMIN_CONNECT to the ResourceOwner role for Topic:${CONNECT_TOPIC}"
+  echo "confluent iam rolebinding create --principal User:$ADMIN_CONNECT --role ResourceOwner --resource Topic:${CONNECT_TOPIC} --kafka-cluster-id $KAFKA_CLUSTER_ID"
+  confluent iam rolebinding create --principal User:$ADMIN_CONNECT --role ResourceOwner --resource Topic:${CONNECT_TOPIC} --kafka-cluster-id $KAFKA_CLUSTER_ID
+done
+
+echo -e "\n# Grant the principal User:$ADMIN_CONNECT to the ResourceOwner role for Group:connect-cluster"
```

Comment:
When the rbac demo will get bigger, we could maybe leverage associative arrays like such:

```sh
resources=(
    'Group:connect-cluster::'
    'Topics:_secrets::(for secret Registry)'
    'Group:secret-registry::(for secret Registry)')

for resource in "${resources[@]}" ; do
    RESOURCE="${resource%%::*}"
    DESCRIPTION="${resource##*::}"

    echo -e "\n# Grant the principal User:$ADMIN_CONNECT to the ResourceOwner role for $RESOURCE $DESCRIPTION"
    echo "confluent iam rolebinding create --principal User:$ADMIN_CONNECT --role ResourceOwner --resource $RESOURCE --kafka-cluster-id $KAFKA_CLUSTER_ID"
    confluent iam rolebinding create --principal User:$ADMIN_CONNECT --role ResourceOwner --resource $RESOURCE --kafka-cluster-id $KAFKA_CLUSTER_ID
done
```

Reply:
@framiere that's very cool bash mojo, but I intentionally didn't make those optimizations for the following reasons:

1. This demo is meant as a learning resource
2. For loops obscure all the required steps...all steps should be unfurled
3. Fancy bash makes it harder for bash newbies to parse
4. Optimizations should be made in a configuration tool

(In fact, I just unfurled my only for loop I had somewhere else)